src/hooks/useResolveEns.ts

```diff
@@ -0,0 +1,31 @@
+import { isAddress } from 'web3-utils'
+import { useState, useEffect } from 'react'
+import { isEns } from 'utils'
+import { resolveENS } from './useSearchSubmit'
+
+export interface ResolvedEns {
+  address: string | null
+  ens?: string
+}
+
+export function useResolveEns(address: string): ResolvedEns | undefined {
+  const [resolvedEns, setResolvedEns] = useState<ResolvedEns | undefined>()
+
+  useEffect(() => {
+    async function _resolveENS(name: string): Promise<void> {
+      const _address = await resolveENS(name)
+      setResolvedEns({ address: _address, ens: name })
+    }
+
+    setResolvedEns(undefined)
+    if (isEns(address)) {
+      _resolveENS(address)
+    } else if (isAddress(address)) {
+      setResolvedEns({ address })
```

Comment:
In this case this is not really an ENS address, is it?

I see what you are trying to do here but maybe the naming is misleading.

Reply:
I have renamed the interface and state properties around [`AddressAccount`](https://github.com/gnosis/gp-ui/blob/869-address-page-by-ens/src/hooks/useResolveEns.ts#L6), what do you think?

Reply:
Looks better, thanks